connectors/airwatch/pom.xml

```diff
@@ -160,6 +161,87 @@
                 </plugins>
             </build>
         </profile>
+
+        <profile>
+            <id>push-docker-image</id>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>io.fabric8</groupId>
+                        <artifactId>docker-maven-plugin</artifactId>
+
+                        <configuration>
+                            <images>
+                                <image>
+                                    <name>airwatch-connector</name>
+                                    <alias>airwatch-connector</alias>
+                                    <build>
+                                        <assembly>
+                                            <inline>
+                                                <files>
+                                                    <file>
+                                                        <source>${project.basedir}/src/main/resources/application.yml</source>
+                                                        <outputDirectory>./</outputDirectory>
+                                                    </file>
+                                                    <file>
+                                                        <source>${project.basedir}/src/test/resources/docker/managed-apps.yml</source>
+                                                        <outputDirectory>./</outputDirectory>
```

Comment:
Please keep in mind that these images must be usable by anyone. We should not be hard-wiring configuration like this.

The piece you are missing is probably that Docker lets you do this:
```
docker run <image> options...
```

So users should be able to do things like:
```
docker run -d -p 8080 ws1connectors/jira-connector --security.oauth2.resource.jwt.key-uri=http://foo
```
This would run the entrypoint, with all the args you've already specified, and then append the user's args. This doesn't seem to work at the moment, though. You might need to switch to `<exec>` instead of `<shell>`, but that's just a guess on my part.
Where the users wants to specify an entire configuration file, they can use volumes or bind mounts to allow the container to access the host file system.

Reply:
Yes.. you are right... I have changed ```shell``` to ```exec```. Now we will be able to specify external configs.

```
docker run -it --mount type=bind,source="/Users/ssathiamoort/projects/github/connectors-workspace-one/connectors/airwatch/src/test/resources/rpmcheck",target=/opt/vmware/airwatch-connector ws1connectors/airwatch-connector /bin/sh --spring.config.location=file:/opt/vmware/airwatch-connector/managed-apps.yml --security.oauth2.resource.jwt.key-uri="https://herocard.vmwareidentity.com/SAAS/API/1.0/REST/auth/token?attribute=publicKey&format=pem" --greenbox.url=https://herocard.vmwareidentity.com
```